File: old/0055_JumpGame.py
class Solution:
    def canJump(self, nums):
        end = len(nums) - 1
        i = len(nums) - 2
        while i >= 0:
            if i + nums[i] >= end:
                end = i
            i -= 1
        if end == 0:
            return True
        else:
            return False

File: top150/array-string/0055_Jump_Game.go
package arraystring

func canJump(nums []int) bool {
	n := len(nums)
	if n <= 1 {
		return true
	}
	target := len(nums) - 1
	for i := n - 2; i >= 0; i-- {
		if i+nums[i] >= target {
			target = i
		}
	}
	if target == 0 {
		return true
	}
	return false
}
